tests/phpunit/Unit/Query/DescriptionFactoryTest.php

```diff
@@ -225,4 +225,35 @@ public function testCanConstructDescriptionFromValidDataValue() {
 		);
 	}
 
+	public function testCanConstructDescriptionFromMonolingualTextValue() {
+
+		$containerSemanticData = $this->getMockBuilder( '\SMWContainerSemanticData' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$containerSemanticData->expects( $this->atLeastOnce() )
+			->method( 'getPropertyValues' )
+			->will( $this->returnValue( array( $this->dataItemFactory->newDIBlob( 'Bar' ) ) ) );
+
+		$dataValue = $this->getMockBuilder( '\SMW\DataValues\MonolingualTextValue' )
+			->disableOriginalConstructor()
+			->setMethods( array( 'isValid', 'getProperty', 'getDataItem' ) )
+			->getMock();
+
+		$dataValue->expects( $this->atLeastOnce() )
+			->method( 'isValid' )
+			->will( $this->returnValue( true ) );
+
+		$dataValue->expects( $this->atLeastOnce() )
+			->method( 'getDataItem' )
+			->will( $this->returnValue( $this->dataItemFactory->newDIContainer( $containerSemanticData ) ) );
+
+		$instance = new DescriptionFactory();
+
+		$this->assertInstanceOf(
+			'SMW\Query\Language\Conjunction',
+			$instance->newFromDataValue( $dataValue )
+		);
+	}
```

Comment:
> This test has 9 string references to code and binds to implementation details.

Yes, that's purpose and if someone breaks the interface the test will fail and that is what I'm counting on. Writing a "real" test double is just not worth it, for this you have the mock object.